Build fails on rolling

<!--
For general questions, please ask on Robotics Stack Exchange: https://robotics.stackexchange.com/, make sure to include at least the `ros2` and `nav2` tags and the rosdistro version you are running, e.g. `ardent`.
For general design discussions, please post on discourse: https://discourse.ros.org/c/ng-ros
Not sure if this is the right repository? Open an issue on https://github.com/ros-navigation/navigation2
For Bug report or feature requests, please fill out the relevant category below
-->

## Bug report

**Required Info:**

- Operating System: Ubuntu 24.04.4 LTS
- Computer: AMD Ryzen 5 3600X 6-Core Processor
- ROS2 Version: Rolling binaries - 0.13.0-1noble.20260214.004037
- Version or commit hash:  cd75eae6c9fd3b7629e2b5f53ed6ffcf80547dcb
- DDS implementation: Fast-RTPS

#### Steps to reproduce issue
<!-- Detailed instructions on how to reliably reproduce this issue http://sscce.org/
``` code that can be copy-pasted is preferred ``` -->
```
colcon build --symlink-install --merge-install  --parallel-workers 6
```

Build fails, here's the logs:
<details>
<summary>nav_2d_utils stderr</summary>

```bash
In file included from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_dwb_controller/nav_2d_utils/test/2d_utils_test.cpp:41:
/opt/ros/rolling/src/gtest_vendor/include/gtest/gtest.h: In instantiation of ‘testing::AssertionResult testing::internal::CmpHelperEQ(const char*, const char*, const T1&, const T2&) [with T1 = builtin_interfaces::msg::Time_<std::allocator<void> >; T2 = rclcpp::Time]’:
/opt/ros/rolling/src/gtest_vendor/include/gtest/gtest.h:1398:23:   required from ‘static testing::AssertionResult testing::internal::EqHelper::Compare(const char*, const char*, const T1&, const T2&) [with T1 = builtin_interfaces::msg::Time_<std::allocator<void> >; T2 = rclcpp::Time; typename std::enable_if<((! std::is_integral<_Tp>::value) || (! std::is_pointer<_Dp>::value))>::type* <anonymous> = 0]’
/home/sayed/Projects/nav2_ws/src/navigation2/nav2_dwb_controller/nav_2d_utils/test/2d_utils_test.cpp:60:3:   required from here
/opt/ros/rolling/src/gtest_vendor/include/gtest/gtest.h:1379:11: error: ambiguous overload for ‘operator==’ (operand types are ‘const builtin_interfaces::msg::Time_<std::allocator<void> >’ and ‘const rclcpp::Time’)
 1379 |   if (lhs == rhs) {
      |       ~~~~^~~~~~
In file included from /opt/ros/rolling/include/rclcpp/rclcpp/clock.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/client.hpp:38,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/callback_group.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/any_executable.hpp:20,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/memory_strategy.hpp:25,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/memory_strategies.hpp:18,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executor_options.hpp:22,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executor.hpp:38,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executors/multi_threaded_executor.hpp:25,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executors.hpp:21,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/rclcpp.hpp:170,
                 from /opt/ros/rolling/include/tf2_ros/tf2_ros/buffer_interface.hpp:50,
                 from /opt/ros/rolling/include/tf2_geometry_msgs/tf2_geometry_msgs/tf2_geometry_msgs.hpp:62,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_dwb_controller/nav_2d_utils/test/2d_utils_test.cpp:39:
/opt/ros/rolling/include/rclcpp/rclcpp/time.hpp:120:3: note: candidate: ‘bool rclcpp::Time::operator==(const rclcpp::Time&) const’ (reversed)
  120 |   operator==(const rclcpp::Time & rhs) const;
      |   ^~~~~~~~
In file included from /opt/ros/rolling/include/std_msgs/std_msgs/msg/detail/header__struct.hpp:24,
                 from /opt/ros/rolling/include/geometry_msgs/geometry_msgs/msg/detail/point_stamped__struct.hpp:24,
                 from /opt/ros/rolling/include/geometry_msgs/geometry_msgs/msg/point_stamped.hpp:7,
                 from /opt/ros/rolling/include/tf2_geometry_msgs/tf2_geometry_msgs/tf2_geometry_msgs.hpp:40,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_dwb_controller/nav_2d_utils/test/2d_utils_test.cpp:39:
/opt/ros/rolling/include/builtin_interfaces/builtin_interfaces/msg/detail/time__struct.hpp:123:8: note: candidate: ‘bool builtin_interfaces::msg::Time_<ContainerAllocator>::operator==(const builtin_interfaces::msg::Time_<ContainerAllocator>&) const [with ContainerAllocator = std::allocator<void>]’
  123 |   bool operator==(const Time_ & other) const
      |        ^~~~~~~~
gmake[2]: *** [test/CMakeFiles/2d_utils_tests.dir/build.make:76: test/CMakeFiles/2d_utils_tests.dir/2d_utils_test.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:283: test/CMakeFiles/2d_utils_tests.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
```

</details>

<details>
<summary>nav2_behavior_tree stderr</summary>

```bash
/home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/plugins/decorator/goal_updater_node.cpp: In member function ‘virtual BT::NodeStatus nav2_behavior_tree::GoalUpdater::tick()’:
/home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/plugins/decorator/goal_updater_node.cpp:97:42: error: ambiguous overload for ‘operator==’ (operand types are ‘std_msgs::msg::Header_<std::allocator<void> >::_stamp_type’ {aka ‘builtin_interfaces::msg::Time_<std::allocator<void> >’} and ‘rclcpp::Time’)
   97 |     if (last_goal_received_.header.stamp == rclcpp::Time(0)) {
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~         ~~~~~~~
      |                                    |                |
      |                                    |                rclcpp::Time
      |                                    std_msgs::msg::Header_<std::allocator<void> >::_stamp_type {aka builtin_interfaces::msg::Time_<std::allocator<void> >}
In file included from /opt/ros/rolling/include/rclcpp/rclcpp/clock.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/client.hpp:38,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/callback_group.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/any_executable.hpp:20,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/memory_strategy.hpp:25,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/memory_strategies.hpp:18,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executor_options.hpp:22,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executor.hpp:38,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executors/multi_threaded_executor.hpp:25,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executors.hpp:21,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/rclcpp.hpp:170,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_ros_common/include/nav2_ros_common/node_utils.hpp:25,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_ros_common/include/nav2_ros_common/lifecycle_node.hpp:26,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/include/nav2_behavior_tree/bt_utils.hpp:22,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/include/nav2_behavior_tree/plugins/decorator/goal_updater_node.hpp:28,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/plugins/decorator/goal_updater_node.cpp:22:
/opt/ros/rolling/include/rclcpp/rclcpp/time.hpp:120:3: note: candidate: ‘bool rclcpp::Time::operator==(const rclcpp::Time&) const’ (reversed)
  120 |   operator==(const rclcpp::Time & rhs) const;
      |   ^~~~~~~~
In file included from /opt/ros/rolling/include/std_msgs/std_msgs/msg/detail/header__struct.hpp:24,
                 from /opt/ros/rolling/include/geometry_msgs/geometry_msgs/msg/detail/pose_stamped__struct.hpp:24,
                 from /opt/ros/rolling/include/geometry_msgs/geometry_msgs/msg/pose_stamped.hpp:7,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/include/nav2_behavior_tree/plugins/decorator/goal_updater_node.hpp:26,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/plugins/decorator/goal_updater_node.cpp:22:
/opt/ros/rolling/include/builtin_interfaces/builtin_interfaces/msg/detail/time__struct.hpp:123:8: note: candidate: ‘bool builtin_interfaces::msg::Time_<ContainerAllocator>::operator==(const builtin_interfaces::msg::Time_<ContainerAllocator>&) const [with ContainerAllocator = std::allocator<void>]’
  123 |   bool operator==(const Time_ & other) const
      |        ^~~~~~~~
/home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/plugins/decorator/goal_updater_node.cpp:122:50: error: ambiguous overload for ‘operator==’ (operand types are ‘std_msgs::msg::Header_<std::allocator<void> >::_stamp_type’ {aka ‘builtin_interfaces::msg::Time_<std::allocator<void> >’} and ‘rclcpp::Time’)
  122 |     } else if (last_goals_received_.header.stamp == rclcpp::Time(0)) {
      |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~         ~~~~~~~
      |                                            |                |
      |                                            |                rclcpp::Time
      |                                            std_msgs::msg::Header_<std::allocator<void> >::_stamp_type {aka builtin_interfaces::msg::Time_<std::allocator<void> >}
In file included from /opt/ros/rolling/include/rclcpp/rclcpp/clock.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/client.hpp:38,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/callback_group.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/any_executable.hpp:20,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/memory_strategy.hpp:25,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/memory_strategies.hpp:18,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executor_options.hpp:22,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executor.hpp:38,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executors/multi_threaded_executor.hpp:25,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executors.hpp:21,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/rclcpp.hpp:170,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_ros_common/include/nav2_ros_common/node_utils.hpp:25,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_ros_common/include/nav2_ros_common/lifecycle_node.hpp:26,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/include/nav2_behavior_tree/bt_utils.hpp:22,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/include/nav2_behavior_tree/plugins/decorator/goal_updater_node.hpp:28,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/plugins/decorator/goal_updater_node.cpp:22:
/opt/ros/rolling/include/rclcpp/rclcpp/time.hpp:120:3: note: candidate: ‘bool rclcpp::Time::operator==(const rclcpp::Time&) const’ (reversed)
  120 |   operator==(const rclcpp::Time & rhs) const;
      |   ^~~~~~~~
In file included from /opt/ros/rolling/include/std_msgs/std_msgs/msg/detail/header__struct.hpp:24,
                 from /opt/ros/rolling/include/geometry_msgs/geometry_msgs/msg/detail/pose_stamped__struct.hpp:24,
                 from /opt/ros/rolling/include/geometry_msgs/geometry_msgs/msg/pose_stamped.hpp:7,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/include/nav2_behavior_tree/plugins/decorator/goal_updater_node.hpp:26,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/plugins/decorator/goal_updater_node.cpp:22:
/opt/ros/rolling/include/builtin_interfaces/builtin_interfaces/msg/detail/time__struct.hpp:123:8: note: candidate: ‘bool builtin_interfaces::msg::Time_<ContainerAllocator>::operator==(const builtin_interfaces::msg::Time_<ContainerAllocator>&) const [with ContainerAllocator = std::allocator<void>]’
  123 |   bool operator==(const Time_ & other) const
      |        ^~~~~~~~
gmake[2]: *** [CMakeFiles/nav2_goal_updater_node_bt_node.dir/build.make:76: CMakeFiles/nav2_goal_updater_node_bt_node.dir/plugins/decorator/goal_updater_node.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:1666: CMakeFiles/nav2_goal_updater_node_bt_node.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
gmake: *** [Makefile:146: all] Error 2



In file included from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/test/plugins/action/test_compute_route_action.cpp:15:
/opt/ros/rolling/src/gtest_vendor/include/gtest/gtest.h: In instantiation of ‘testing::AssertionResult testing::internal::CmpHelperEQ(const char*, const char*, const T1&, const T2&) [with T1 = builtin_interfaces::msg::Duration_<std::allocator<void> >; T2 = rclcpp::Duration]’:
/opt/ros/rolling/src/gtest_vendor/include/gtest/gtest.h:1398:23:   required from ‘static testing::AssertionResult testing::internal::EqHelper::Compare(const char*, const char*, const T1&, const T2&) [with T1 = builtin_interfaces::msg::Duration_<std::allocator<void> >; T2 = rclcpp::Duration; typename std::enable_if<((! std::is_integral<_Tp>::value) || (! std::is_pointer<_Dp>::value))>::type* <anonymous> = 0]’
/home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/test/plugins/action/test_compute_route_action.cpp:166:3:   required from here
/opt/ros/rolling/src/gtest_vendor/include/gtest/gtest.h:1379:11: error: ambiguous overload for ‘operator==’ (operand types are ‘const builtin_interfaces::msg::Duration_<std::allocator<void> >’ and ‘const rclcpp::Duration’)
 1379 |   if (lhs == rhs) {
      |       ~~~~^~~~~~
In file included from /opt/ros/rolling/include/rclcpp/rclcpp/time.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/clock.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/client.hpp:38,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/callback_group.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/any_executable.hpp:20,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/memory_strategy.hpp:25,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/memory_strategies.hpp:18,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executor_options.hpp:22,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executor.hpp:38,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executors/multi_threaded_executor.hpp:25,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executors.hpp:21,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/rclcpp.hpp:170,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/include/nav2_behavior_tree/utils/test_action_server.hpp:21,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/test/plugins/action/test_compute_route_action.cpp:25:
/opt/ros/rolling/include/rclcpp/rclcpp/duration.hpp:76:3: note: candidate: ‘bool rclcpp::Duration::operator==(const rclcpp::Duration&) const’ (reversed)
   76 |   operator==(const rclcpp::Duration & rhs) const;
      |   ^~~~~~~~
In file included from /opt/ros/rolling/include/builtin_interfaces/builtin_interfaces/msg/duration.hpp:7,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/duration.hpp:20,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/time.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/clock.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/client.hpp:38,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/callback_group.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/any_executable.hpp:20,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/memory_strategy.hpp:25,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/memory_strategies.hpp:18,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executor_options.hpp:22,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executor.hpp:38,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executors/multi_threaded_executor.hpp:25,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executors.hpp:21,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/rclcpp.hpp:170,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/include/nav2_behavior_tree/utils/test_action_server.hpp:21,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/test/plugins/action/test_compute_route_action.cpp:25:
/opt/ros/rolling/include/builtin_interfaces/builtin_interfaces/msg/detail/duration__struct.hpp:123:8: note: candidate: ‘bool builtin_interfaces::msg::Duration_<ContainerAllocator>::operator==(const builtin_interfaces::msg::Duration_<ContainerAllocator>&) const [with ContainerAllocator = std::allocator<void>]’
  123 |   bool operator==(const Duration_ & other) const
      |        ^~~~~~~~
In file included from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/test/plugins/action/test_compute_and_track_route_action.cpp:15:
/opt/ros/rolling/src/gtest_vendor/include/gtest/gtest.h: In instantiation of ‘testing::AssertionResult testing::internal::CmpHelperEQ(const char*, const char*, const T1&, const T2&) [with T1 = builtin_interfaces::msg::Duration_<std::allocator<void> >; T2 = rclcpp::Duration]’:
/opt/ros/rolling/src/gtest_vendor/include/gtest/gtest.h:1398:23:   required from ‘static testing::AssertionResult testing::internal::EqHelper::Compare(const char*, const char*, const T1&, const T2&) [with T1 = builtin_interfaces::msg::Duration_<std::allocator<void> >; T2 = rclcpp::Duration; typename std::enable_if<((! std::is_integral<_Tp>::value) || (! std::is_pointer<_Dp>::value))>::type* <anonymous> = 0]’
/home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/test/plugins/action/test_compute_and_track_route_action.cpp:157:3:   required from here
/opt/ros/rolling/src/gtest_vendor/include/gtest/gtest.h:1379:11: error: ambiguous overload for ‘operator==’ (operand types are ‘const builtin_interfaces::msg::Duration_<std::allocator<void> >’ and ‘const rclcpp::Duration’)
 1379 |   if (lhs == rhs) {
      |       ~~~~^~~~~~
In file included from /opt/ros/rolling/include/rclcpp/rclcpp/time.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/clock.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/client.hpp:38,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/callback_group.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/any_executable.hpp:20,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/memory_strategy.hpp:25,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/memory_strategies.hpp:18,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executor_options.hpp:22,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executor.hpp:38,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executors/multi_threaded_executor.hpp:25,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executors.hpp:21,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/rclcpp.hpp:170,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/include/nav2_behavior_tree/utils/test_action_server.hpp:21,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/test/plugins/action/test_compute_and_track_route_action.cpp:25:
/opt/ros/rolling/include/rclcpp/rclcpp/duration.hpp:76:3: note: candidate: ‘bool rclcpp::Duration::operator==(const rclcpp::Duration&) const’ (reversed)
   76 |   operator==(const rclcpp::Duration & rhs) const;
      |   ^~~~~~~~
In file included from /opt/ros/rolling/include/builtin_interfaces/builtin_interfaces/msg/duration.hpp:7,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/duration.hpp:20,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/time.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/clock.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/client.hpp:38,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/callback_group.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/any_executable.hpp:20,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/memory_strategy.hpp:25,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/memory_strategies.hpp:18,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executor_options.hpp:22,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executor.hpp:38,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executors/multi_threaded_executor.hpp:25,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/executors.hpp:21,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/rclcpp.hpp:170,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/include/nav2_behavior_tree/utils/test_action_server.hpp:21,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_behavior_tree/test/plugins/action/test_compute_and_track_route_action.cpp:25:
/opt/ros/rolling/include/builtin_interfaces/builtin_interfaces/msg/detail/duration__struct.hpp:123:8: note: candidate: ‘bool builtin_interfaces::msg::Duration_<ContainerAllocator>::operator==(const builtin_interfaces::msg::Duration_<ContainerAllocator>&) const [with ContainerAllocator = std::allocator<void>]’
  123 |   bool operator==(const Duration_ & other) const
      |        ^~~~~~~~
gmake[2]: *** [test/plugins/action/CMakeFiles/test_action_compute_route_action.dir/build.make:76: test/plugins/action/CMakeFiles/test_action_compute_route_action.dir/test_compute_route_action.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:4135: test/plugins/action/CMakeFiles/test_action_compute_route_action.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
gmake[2]: *** [test/plugins/action/CMakeFiles/test_action_compute_and_track_route_action.dir/build.make:76: test/plugins/action/CMakeFiles/test_action_compute_and_track_route_action.dir/test_compute_and_track_route_action.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:4164: test/plugins/action/CMakeFiles/test_action_compute_and_track_route_action.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2

```

</details>


<details>
<summary>opennav_following stderr</summary>

```bash

/home/sayed/Projects/nav2_ws/src/navigation2/nav2_following/opennav_following/src/following_server.cpp: In member function ‘virtual bool opennav_following::FollowingServer::getRefinedPose(geometry_msgs::msg::PoseStamped&)’:
/home/sayed/Projects/nav2_ws/src/navigation2/nav2_following/opennav_following/src/following_server.cpp:526:29: error: ambiguous overload for ‘operator==’ (operand types are ‘std_msgs::msg::Header_<std::allocator<void> >::_stamp_type’ {aka ‘builtin_interfaces::msg::Time_<std::allocator<void> >’} and ‘rclcpp::Time’)
  526 |   if (detected.header.stamp == rclcpp::Time(0)) {
      |       ~~~~~~~~~~~~~~~~~~~~~ ^~         ~~~~~~~
      |                       |                |
      |                       |                rclcpp::Time
      |                       std_msgs::msg::Header_<std::allocator<void> >::_stamp_type {aka builtin_interfaces::msg::Time_<std::allocator<void> >}
In file included from /opt/ros/rolling/include/rclcpp/rclcpp/clock.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/rate.hpp:22,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_ros_common/include/nav2_ros_common/rate.hpp:23,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_following/opennav_following/src/following_server.cpp:17:
/opt/ros/rolling/include/rclcpp/rclcpp/time.hpp:120:3: note: candidate: ‘bool rclcpp::Time::operator==(const rclcpp::Time&) const’ (reversed)
  120 |   operator==(const rclcpp::Time & rhs) const;
      |   ^~~~~~~~
In file included from /opt/ros/rolling/include/builtin_interfaces/builtin_interfaces/msg/time.hpp:7,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/time.hpp:18,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/clock.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/rate.hpp:22,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_ros_common/include/nav2_ros_common/rate.hpp:23,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_following/opennav_following/src/following_server.cpp:17:
/opt/ros/rolling/include/builtin_interfaces/builtin_interfaces/msg/detail/time__struct.hpp:123:8: note: candidate: ‘bool builtin_interfaces::msg::Time_<ContainerAllocator>::operator==(const builtin_interfaces::msg::Time_<ContainerAllocator>&) const [with ContainerAllocator = std::allocator<void>]’
  123 |   bool operator==(const Time_ & other) const
      |        ^~~~~~~~
/home/sayed/Projects/nav2_ws/src/navigation2/nav2_following/opennav_following/src/following_server.cpp:532:47: error: ambiguous overload for ‘operator!=’ (operand types are ‘std_msgs::msg::Header_<std::allocator<void> >::_stamp_type’ {aka ‘builtin_interfaces::msg::Time_<std::allocator<void> >’} and ‘rclcpp::Time’)
  532 |       if (detected_dynamic_pose_.header.stamp != rclcpp::Time(0)) {
      |           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~         ~~~~~~~
      |                                         |                |
      |                                         |                rclcpp::Time
      |                                         std_msgs::msg::Header_<std::allocator<void> >::_stamp_type {aka builtin_interfaces::msg::Time_<std::allocator<void> >}
In file included from /opt/ros/rolling/include/rclcpp/rclcpp/clock.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/rate.hpp:22,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_ros_common/include/nav2_ros_common/rate.hpp:23,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_following/opennav_following/src/following_server.cpp:17:
/opt/ros/rolling/include/rclcpp/rclcpp/time.hpp:120:3: note: candidate: ‘bool rclcpp::Time::operator==(const rclcpp::Time&) const’ (reversed)
  120 |   operator==(const rclcpp::Time & rhs) const;
      |   ^~~~~~~~
In file included from /opt/ros/rolling/include/builtin_interfaces/builtin_interfaces/msg/time.hpp:7,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/time.hpp:18,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/clock.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/rate.hpp:22,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_ros_common/include/nav2_ros_common/rate.hpp:23,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_following/opennav_following/src/following_server.cpp:17:
/opt/ros/rolling/include/builtin_interfaces/builtin_interfaces/msg/detail/time__struct.hpp:123:8: note: candidate: ‘bool builtin_interfaces::msg::Time_<ContainerAllocator>::operator==(const builtin_interfaces::msg::Time_<ContainerAllocator>&) const [with ContainerAllocator = std::allocator<void>]’ (rewritten)
  123 |   bool operator==(const Time_ & other) const
      |        ^~~~~~~~
/opt/ros/rolling/include/builtin_interfaces/builtin_interfaces/msg/detail/time__struct.hpp:133:8: note: candidate: ‘bool builtin_interfaces::msg::Time_<ContainerAllocator>::operator!=(const builtin_interfaces::msg::Time_<ContainerAllocator>&) const [with ContainerAllocator = std::allocator<void>]’
  133 |   bool operator!=(const Time_ & other) const
      |        ^~~~~~~~
/home/sayed/Projects/nav2_ws/src/navigation2/nav2_following/opennav_following/src/following_server.cpp:538:31: error: ambiguous overload for ‘operator==’ (operand types are ‘std_msgs::msg::Header_<std::allocator<void> >::_stamp_type’ {aka ‘builtin_interfaces::msg::Time_<std::allocator<void> >’} and ‘rclcpp::Time’)
  538 |     if (detected.header.stamp == rclcpp::Time(0)) {
      |         ~~~~~~~~~~~~~~~~~~~~~ ^~         ~~~~~~~
      |                         |                |
      |                         |                rclcpp::Time
      |                         std_msgs::msg::Header_<std::allocator<void> >::_stamp_type {aka builtin_interfaces::msg::Time_<std::allocator<void> >}
In file included from /opt/ros/rolling/include/rclcpp/rclcpp/clock.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/rate.hpp:22,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_ros_common/include/nav2_ros_common/rate.hpp:23,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_following/opennav_following/src/following_server.cpp:17:
/opt/ros/rolling/include/rclcpp/rclcpp/time.hpp:120:3: note: candidate: ‘bool rclcpp::Time::operator==(const rclcpp::Time&) const’ (reversed)
  120 |   operator==(const rclcpp::Time & rhs) const;
      |   ^~~~~~~~
In file included from /opt/ros/rolling/include/builtin_interfaces/builtin_interfaces/msg/time.hpp:7,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/time.hpp:18,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/clock.hpp:24,
                 from /opt/ros/rolling/include/rclcpp/rclcpp/rate.hpp:22,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_ros_common/include/nav2_ros_common/rate.hpp:23,
                 from /home/sayed/Projects/nav2_ws/src/navigation2/nav2_following/opennav_following/src/following_server.cpp:17:
/opt/ros/rolling/include/builtin_interfaces/builtin_interfaces/msg/detail/time__struct.hpp:123:8: note: candidate: ‘bool builtin_interfaces::msg::Time_<ContainerAllocator>::operator==(const builtin_interfaces::msg::Time_<ContainerAllocator>&) const [with ContainerAllocator = std::allocator<void>]’
  123 |   bool operator==(const Time_ & other) const
      |        ^~~~~~~~
gmake[2]: *** [CMakeFiles/opennav_following_core.dir/build.make:90: CMakeFiles/opennav_following_core.dir/src/following_server.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:175: CMakeFiles/opennav_following_core.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2

```
</details>

Related to #6111 

CC @mini-1235

--- a/nav2_behavior_tree/plugins/decorator/goal_updater_node.cpp
+++ b/nav2_behavior_tree/plugins/decorator/goal_updater_node.cpp
@@ -94,7 +94,7 @@ inline BT::NodeStatus GoalUpdater::tick()
   callback_group_executor_.spin_all(bt_loop_duration_);
 
   if (last_goal_received_set_) {
-    if (rclcpp::Time(last_goal_received_.header.stamp, RCL_ROS_TIME) == rclcpp::Time(0)) {
+    if (last_goal_received_.header.stamp == builtin_interfaces::msg::Time{}) {
       // if the goal doesn't have a timestamp, we reject it
       RCLCPP_WARN(
         node_->get_logger(), "The received goal has no timestamp. Ignoring.");
@@ -119,7 +119,7 @@ inline BT::NodeStatus GoalUpdater::tick()
   if (last_goals_received_set_) {
     if (last_goals_received_.goals.empty()) {
       setOutput("output_goals", goals);
-    } else if (rclcpp::Time(last_goals_received_.header.stamp, RCL_ROS_TIME) == rclcpp::Time(0)) {
+    } else if (last_goals_received_.header.stamp == builtin_interfaces::msg::Time{}) {
       RCLCPP_WARN(
         node_->get_logger(), "The received goals array has no timestamp. Ignoring.");
       setOutput("output_goals", goals);

--- a/nav2_behavior_tree/plugins/decorator/goal_updater_node.cpp
+++ b/nav2_behavior_tree/plugins/decorator/goal_updater_node.cpp
@@ -94,7 +94,7 @@ inline BT::NodeStatus GoalUpdater::tick()
   callback_group_executor_.spin_all(bt_loop_duration_);
 
   if (last_goal_received_set_) {
-    if (last_goal_received_.header.stamp == rclcpp::Time(0)) {
+    if (rclcpp::Time(last_goal_received_.header.stamp, RCL_ROS_TIME) == rclcpp::Time(0)) {
       // if the goal doesn't have a timestamp, we reject it
       RCLCPP_WARN(
         node_->get_logger(), "The received goal has no timestamp. Ignoring.");
@@ -119,7 +119,7 @@ inline BT::NodeStatus GoalUpdater::tick()
   if (last_goals_received_set_) {
     if (last_goals_received_.goals.empty()) {
       setOutput("output_goals", goals);
-    } else if (last_goals_received_.header.stamp == rclcpp::Time(0)) {
+    } else if (rclcpp::Time(last_goals_received_.header.stamp, RCL_ROS_TIME) == rclcpp::Time(0)) {
       RCLCPP_WARN(
         node_->get_logger(), "The received goals array has no timestamp. Ignoring.");
       setOutput("output_goals", goals);

--- a/nav2_behavior_tree/test/plugins/action/test_compute_and_track_route_action.cpp
+++ b/nav2_behavior_tree/test/plugins/action/test_compute_and_track_route_action.cpp
@@ -154,7 +154,7 @@ TEST_F(ComputeAndTrackRouteActionTestFixture, test_tick_poses)
   builtin_interfaces::msg::Duration time1;
   EXPECT_TRUE(
     config_->blackboard->get<builtin_interfaces::msg::Duration>("execution_duration", time1));
-  EXPECT_EQ(time1, rclcpp::Duration::from_seconds(0.1));
+  EXPECT_EQ(rclcpp::Duration(time1), rclcpp::Duration::from_seconds(0.1));
 
   // halt node so another goal can be sent
   tree_->haltTree();
@@ -202,7 +202,7 @@ TEST_F(ComputeAndTrackRouteActionTestFixture, test_tick_ids)
   builtin_interfaces::msg::Duration time1;
   EXPECT_TRUE(
     config_->blackboard->get<builtin_interfaces::msg::Duration>("execution_duration", time1));
-  EXPECT_EQ(time1, rclcpp::Duration::from_seconds(0.1));
+  EXPECT_EQ(rclcpp::Duration(time1), rclcpp::Duration::from_seconds(0.1));
 
   // halt node so another goal can be sent
   tree_->haltTree();

--- a/nav2_behavior_tree/test/plugins/action/test_compute_route_action.cpp
+++ b/nav2_behavior_tree/test/plugins/action/test_compute_route_action.cpp
@@ -163,7 +163,7 @@ TEST_F(ComputeRouteActionTestFixture, test_IDs)
 
   builtin_interfaces::msg::Duration time1;
   EXPECT_TRUE(config_->blackboard->get<builtin_interfaces::msg::Duration>("planning_time", time1));
-  EXPECT_EQ(time1, rclcpp::Duration::from_seconds(0.1));
+  EXPECT_EQ(rclcpp::Duration(time1), rclcpp::Duration::from_seconds(0.1));
 
   // halt node so another goal can be sent
   tree_->haltTree();
@@ -221,7 +221,7 @@ TEST_F(ComputeRouteActionTestFixture, test_poses)
 
   builtin_interfaces::msg::Duration time1;
   EXPECT_TRUE(config_->blackboard->get<builtin_interfaces::msg::Duration>("planning_time", time1));
-  EXPECT_EQ(time1, rclcpp::Duration::from_seconds(0.1));
+  EXPECT_EQ(rclcpp::Duration(time1), rclcpp::Duration::from_seconds(0.1));
 
   // halt node so another goal can be sent
   tree_->haltTree();

--- a/nav2_dwb_controller/nav_2d_utils/test/2d_utils_test.cpp
+++ b/nav2_dwb_controller/nav_2d_utils/test/2d_utils_test.cpp
@@ -57,7 +57,7 @@ TEST(nav_2d_utils, PosesToPathEmpty)
 
   EXPECT_EQ(path.poses.size(), 0ul);
   EXPECT_EQ(path.header.frame_id, frame);
-  EXPECT_EQ(path.header.stamp, time);
+  EXPECT_EQ(rclcpp::Time(path.header.stamp, RCL_ROS_TIME), time);
 }
 
 TEST(nav_2d_utils, PosesToPathNonEmpty)
@@ -92,11 +92,11 @@ TEST(nav_2d_utils, PosesToPathNonEmpty)
 
   for (const auto & stamped_pose : path.poses) {
     EXPECT_EQ(stamped_pose.header.frame_id, frame);
-    EXPECT_EQ(stamped_pose.header.stamp, time);
+    EXPECT_EQ(rclcpp::Time(stamped_pose.header.stamp, RCL_ROS_TIME), time);
   }
 
   EXPECT_EQ(path.header.frame_id, frame);
-  EXPECT_EQ(path.header.stamp, time);
+  EXPECT_EQ(rclcpp::Time(path.header.stamp, RCL_ROS_TIME), time);
 }
 
 

--- a/nav2_following/opennav_following/src/following_server.cpp
+++ b/nav2_following/opennav_following/src/following_server.cpp
@@ -523,19 +523,19 @@ bool FollowingServer::getRefinedPose(geometry_msgs::msg::PoseStamped & pose)
   geometry_msgs::msg::PoseStamped detected = detected_dynamic_pose_;
 
   // If we haven't received any detection yet, wait up to detection_timeout_ for one to arrive.
-  if (detected.header.stamp == rclcpp::Time(0)) {
+  if (rclcpp::Time(detected.header.stamp, RCL_ROS_TIME) == rclcpp::Time(0)) {
     auto start = this->now();
     auto timeout = rclcpp::Duration::from_seconds(params_->detection_timeout);
     nav2::Rate wait_rate(this, params_->controller_frequency);
     while (this->now() - start < timeout) {
       // Check if a new detection arrived
-      if (detected_dynamic_pose_.header.stamp != rclcpp::Time(0)) {
+      if (rclcpp::Time(detected_dynamic_pose_.header.stamp, RCL_ROS_TIME) != rclcpp::Time(0)) {
         detected = detected_dynamic_pose_;
         break;
       }
       wait_rate.sleep();
     }
-    if (detected.header.stamp == rclcpp::Time(0)) {
+    if (rclcpp::Time(detected.header.stamp, RCL_ROS_TIME) == rclcpp::Time(0)) {
       RCLCPP_WARN(this->get_logger(), "No detection received within timeout period");
       return false;
     }

--- a/nav2_following/opennav_following/src/following_server.cpp
+++ b/nav2_following/opennav_following/src/following_server.cpp
@@ -523,19 +523,19 @@ bool FollowingServer::getRefinedPose(geometry_msgs::msg::PoseStamped & pose)
   geometry_msgs::msg::PoseStamped detected = detected_dynamic_pose_;
 
   // If we haven't received any detection yet, wait up to detection_timeout_ for one to arrive.
-  if (rclcpp::Time(detected.header.stamp, RCL_ROS_TIME) == rclcpp::Time(0)) {
+  if (detected.header.stamp == builtin_interfaces::msg::Time{}) {
     auto start = this->now();
     auto timeout = rclcpp::Duration::from_seconds(params_->detection_timeout);
     nav2::Rate wait_rate(this, params_->controller_frequency);
     while (this->now() - start < timeout) {
       // Check if a new detection arrived
-      if (rclcpp::Time(detected_dynamic_pose_.header.stamp, RCL_ROS_TIME) != rclcpp::Time(0)) {
+      if (detected_dynamic_pose_.header.stamp != builtin_interfaces::msg::Time{}) {
         detected = detected_dynamic_pose_;
         break;
       }
       wait_rate.sleep();
     }
-    if (rclcpp::Time(detected.header.stamp, RCL_ROS_TIME) == rclcpp::Time(0)) {
+    if (detected.header.stamp == builtin_interfaces::msg::Time{}) {
       RCLCPP_WARN(this->get_logger(), "No detection received within timeout period");
       return false;
     }

